app/src/main/java/org/mozilla/fenix/settings/logins/SaveLoginSettingFragment.kt

```diff
@@ -0,0 +1,43 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+package org.mozilla.fenix.settings.logins
+
+import android.os.Bundle
+import androidx.preference.PreferenceFragmentCompat
+import org.mozilla.fenix.R
+import org.mozilla.fenix.ext.showToolbar
+import org.mozilla.fenix.settings.RadioButtonPreference
+
+class SaveLoginSettingFragment : PreferenceFragmentCompat() {
+    private lateinit var radioSave: RadioButtonPreference
+    private lateinit var radioNeverSave: RadioButtonPreference
+
+    override fun onCreatePreferences(savedInstanceState: Bundle?, rootKey: String?) {
+        setPreferencesFromResource(R.xml.save_logins_preferences, rootKey)
+    }
+
+    override fun onResume() {
+        super.onResume()
+        showToolbar(getString(R.string.preferences_passwords_save_logins))
+        bindSave()
+        bindNeverSave()
+        setupRadioGroups()
```

Comment:
This is a simple enough example that it's not scary here, but there's an important concept here that can make code a lot more difficult to work with as it grows.  

This method will fail if these methods are called out of order (they're ["temporally coupled"](https://www.pluralsight.com/tech-blog/forms-of-temporal-coupling/#sequencing), because everything in programming needs an intimidating name), but there's nothing here telling the compiler (or the reader) that that will happen.  So we need to either read through the method implementations, or just poke around at runtime to find that out.

The easiest way to clean up this problem is, wherever possible, to avoid having methods reach outside of their own scope.  In this case, that's really easy to do.  `bindSave` and `bindNeverSave` can return their buttons, and `setupRadioGroups` can accept them as parameters.  Then the compiler will enforce that they're called in order, and whoever is reading the code isn't forced to keep class state in mind while reading a method.

That could look something like this (names and structure could be played with a bit, but you get the point):

```kotlin
override fun onResume() {
    super.onResume()
    showToolbar(getString(R.string.preferences_passwords_save_logins))
    val save = bindSave()
    val neverSave = bindNeverSave()
    setupRadioGroups(save, neverSave)
}

private fun bindSave(): RadioButtonPreference {
    val keyStrict = getString(R.string.pref_key_save_logins)
    return requireNotNull(findPreference(keyStrict))
}

private fun bindNeverSave(): RadioButtonPreference {
    val keyStandard = getString(R.string.pref_key_never_save_logins)
    return requireNotNull(findPreference(keyStandard))
}

private fun setupRadioGroups(radioNeverSave: RadioButtonPreference, radioSave: RadioButtonPreference) {
    radioNeverSave.addToRadioGroup(radioSave)
    radioSave.addToRadioGroup(radioNeverSave)
}
```